### PR TITLE
Add hierarchy checks to `[p]warn`

### PR DESCRIPTION
**Pull request by kingsleyzhong**
### Type

- [X] Bugfix
- [ ] Enhancement
- [ ] New feature

### Description of the changes

Added checks to warnings to prevent abuse.

### CODE REVIEW

File: redbot/cogs/warnings/warnings.py [hunk 1]
```diff
@@ -346,6 +346,12 @@ async def warn(
         if user.bot:
             await ctx.send(_("You cannot warn other bots."))
             return
+        if user == ctx.guild.owner:
+            await ctx.send(_("You cannot warn the server owner."))
+            return
+        if user.top_role >= ctx.author.top_role and ctx.author != ctx.guild.owner:
+            await ctx.send(_("The person you're trying to warn is equal or higher to you on the discord hierarchy, you cannot warn them."))
```
Comment (Dav-Git) [inline redbot/cogs/warnings/warnings.py]: I'm pretty sure it should be "is equal or higher than you in the discord hierarchy"

**Review by Dav-Git — CHANGES_REQUESTED** [review]
You also need to format this using black. See https://github.com/Cog-Creators/Red-DiscordBot/blob/V3/develop/CONTRIBUTING.md

**Review by Drapersniper — DISMISSED** [review]
Both comments from Dav can be considered requested changes for this PR.

**Review by Kowlin — COMMENTED** [review]
As an additional note to add on-top of the already existing comments, Style checks are failing. You can run the style checking from the make file.

**Review by Kowlin — APPROVED** [review]

**Comment by Kowlin** [comment]
Thank you for your contribution.

**Pull request merged**